File: <100/031/031.py
def main() -> None:
    dp: list[int]    = [0 for _ in range(210)]
    coins: list[int] = [1, 2, 5, 10, 20, 50, 100, 200]
    target: int = 200

    dp[0] = 1
    for coin in coins:
        for m in range(target):
            nm: int = m + coin
            if nm > target: continue
            dp[nm] += dp[m]
    
    print(dp[target])
    return


if __name__ == '__main__':
    main()

File: <100/031/031.cpp
#include<iostream>
using namespace std;

int dp[210], coins[] = {1, 2, 5, 10, 20, 50, 100, 200};
const int target = 200, N = 8;
int main(){
  dp[0] = 1;
  for(int i = 0; i < N; ++i){
    for(int m = 0; m < target; ++m){
      int nm = m + coins[i];
      if(nm > target) break;
      dp[nm] += dp[m];
    }
  }

  cout << dp[target] << endl;
  return 0;
}